src/django_mysql/models/functions.py

```diff
@@ -118,6 +119,25 @@ def __init__(
         super().__init__(field, *values_exprs)
 
 
+class NaturalSortKey(Func):
+    function = "NATURAL_SORT_KEY"
+
+    def __init__(self, expression: ExpressionArgument) -> None:
+        super().__init__(expression)
+
+    def as_sql(
+        self,
+        compiler: SQLCompiler,
+        connection: BaseDatabaseWrapper,
+        **extra_context,
+    ) -> tuple[str, tuple[Any, ...]]:
+        if connection.vendor == "mysql" and not connection.mysql_is_mariadb:
```

Comment:
These database version checks are not worth it. In the successful path, all of this checking is time wasted (creating a `Version` is not free!). In the incredibly rare failure path, they’re not really providing more information than the database’s own error (“no such function”) would provide.

This is why none of the other functions have such checks.

Reply:
Thank you for your feedback. I understand your concerns about the database version checks in the code. But my reasoning was that While these checks can introduce overhead and may not always provide significant additional information, they serve as a preventive measure to catch potential issues early on. 

I will look through the open issue pool on the Django projects to find a first-time-contributor friendly to contribute to.

Reply:
Yeah, I'm all for preventative checks, but their cost has to be balanced against the probability of making a given mistake.